sklearn/ensemble/_gb.py

```diff
@@ -344,6 +346,10 @@ def _clear_state(self):
             del self.train_score_
         if hasattr(self, "oob_improvement_"):
             del self.oob_improvement_
+        if hasattr(self, "oob_scores_ "):
+            del self.oob_scores_
```

Comment:
It seems not covered by the test. We should have a check when we call `fit` that the id of those arrays changes since we delete the object.

Reply:
I have updated the test to check if the id has changed after calling `fit`, codecov still flags it for some reason.